velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1213,6 +1214,20 @@ struct CurrentDateFunction {
   }
 };
 
+template <typename T>
+struct FromISO8601DateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Date>& result,
+      const arg_type<Varchar>& isoDateStr) {
+    int len = std::min((int)isoDateStr.size(), 10);
```

Comment:
Could you try : 
result = len < 10 ? DATE()->toDays(isoDateStr) : DATE()->toDays(isoDateStr->getString().substr(0, 10)); ?

and remove the isoDateExtracted variable assignment.

Reply:
Modified as such, removed the variable assignment and condensed the operations.